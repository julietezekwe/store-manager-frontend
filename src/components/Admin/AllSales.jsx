import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getAllSales } from '../../actions/salesAction';

export class AllSales extends Component {
  async componentDidMount() {
    await this.props.getAllSales();
  }

  render() {
    const { sales } = this.props.sales;
    return (
      <React.Fragment>
        <div className="col-8">
          <div className="center">
            <div className="col-3 " id="totalProducts">
              Total Products Sold: {sales.totalQuantitySold}
            </div>
            <div className="col-4" id="totalPrice">
              Amount Realized: &#8358; {sales.totalAmount}
            </div>
            <div className="col-3" id="totalSales">
              Unique Products No: {sales.totalProductsSold}
            </div>
          </div>

          <div className="card col-11">
            <table className="table">
              <thead>
                <tr>
                  <th>Product Id</th>
                  <th>Product Name</th>
                  <th>Description</th>
                  <th>Quantity</th>
                  <th>Price</th>
                  <th>Seller</th>
                </tr>
              </thead>
              <tbody>
                {sales.SalesModel
                  && sales.SalesModel.map((sale, i) => (
                    <tr key={i}>
                      <td>{sale.productid}</td>
                      <td>{sale.productname}</td>
                      <td>{sale.description}</td>
                      <td>{sale.quantity}</td>
                      <td>{sale.price}</td>
                      <td>{sale.name}</td>
                    </tr>
                  ))}
              </tbody>
            </table>
          </div>
        </div>
      </React.Fragment>
    );
  }
}
const mapStateToProps = state => ({
  sales: state.sales,
});
export default connect(
  mapStateToProps,
  { getAllSales },
)(AllSales);
