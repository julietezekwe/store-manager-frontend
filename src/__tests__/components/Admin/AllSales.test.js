/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import { AllSales } from '../../../components/Admin/AllSales.jsx';

let wrapper;
const props = {
  getAllSales: () => Promise.resolve(),
  sales: {
    sales: {
      totatQuantitySold: 1,
      totalAmount: 10,
      totalProductsSold: 199,
      SalesModel: [{
        productid: 1,
        productname: 'shoes',
        description: 'fine',
        quantity: 10,
        price: 1200,
        name: 'oge',
      }],
    },
  },
};
beforeEach(() => {
  wrapper = mount(<AllSales {...props} />);
});
describe('App Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
