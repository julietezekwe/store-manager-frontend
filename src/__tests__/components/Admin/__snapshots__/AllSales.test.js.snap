// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match snapshot 1`] = `
<AllSales
  getAllSales={[Function]}
  sales={
    Object {
      "sales": Object {
        "SalesModel": Array [
          Object {
            "description": "fine",
            "name": "oge",
            "price": 1200,
            "productid": 1,
            "productname": "shoes",
            "quantity": 10,
          },
        ],
        "totalAmount": 10,
        "totalProductsSold": 199,
        "totatQuantitySold": 1,
      },
    }
  }
>
  <div
    className="col-8"
  >
    <div
      className="center"
    >
      <div
        className="col-3 "
        id="totalProducts"
      >
        Total Products Sold: 
      </div>
      <div
        className="col-4"
        id="totalPrice"
      >
        Amount Realized: â‚¦ 
        10
      </div>
      <div
        className="col-3"
        id="totalSales"
      >
        Unique Products No: 
        199
      </div>
    </div>
    <div
      className="card col-11"
    >
      <table
        className="table"
      >
        <thead>
          <tr>
            <th>
              Product Id
            </th>
            <th>
              Product Name
            </th>
            <th>
              Description
            </th>
            <th>
              Quantity
            </th>
            <th>
              Price
            </th>
            <th>
              Seller
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            key="0"
          >
            <td>
              1
            </td>
            <td>
              shoes
            </td>
            <td>
              fine
            </td>
            <td>
              10
            </td>
            <td>
              1200
            </td>
            <td>
              oge
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</AllSales>
`;
